test/sqllogictest/string.slt

```diff
@@ -863,3 +863,105 @@ SELECT ''::pg_catalog.CHAR(0)
 
 query error length for type char must be within \[1-10485760\], have 10485761
 SELECT ''::pg_catalog.CHAR(10485761)
+
+### right ###
```

Comment:
Sorry, I only figured this out after the first review/talking to @quodlibetor, but could you also add a test for [combining characters](https://en.wikipedia.org/wiki/Combining_character#:~:text=In%20digital%20typography%2C%20combining%20characters,marks%20(including%20combining%20accents).)? We only need to match Postgres' behavior here (which is a bit wonky):
```
jessicalaughlin=# select right('\u0930\u0903\u0951', 0);
 right
-------

(1 row)

jessicalaughlin=# select right('\u0930\u0903\u0951', 1);
 right
-------
 1
(1 row)

jessicalaughlin=# select right('\u0930\u0903\u0951', -1);
       right
-------------------
 u0930\u0903\u0951
```

Reply:
adding a test!